android/engine/src/main/java/org/smartregister/fhircore/engine/ui/register/BaseRegisterActivity.kt

```diff
@@ -698,6 +695,29 @@ abstract class BaseRegisterActivity :
 
   open fun onBarcodeResult(barcode: String, view: View) {}
 
+  private fun handleSyncFailed(state: State) {
+
+    val exceptions =
+      when (state) {
+        is State.Glitch -> state.exceptions
+        is State.Failed -> state.result.exceptions
+        else -> listOf()
+      }
+
+    if (exceptions.map { it.exception }.filterIsInstance<HttpException>().firstOrNull()?.code() ==
+        401
+    ) {
+      showToast(getString(R.string.session_expired))
+      accountAuthenticator.logout()
```

Comment:
This is an issue in the current OpenSRP. Hold off on this awaiting discussions within OpenSRP

Reply:
See this for more information https://github.com/opensrp/support/issues/108